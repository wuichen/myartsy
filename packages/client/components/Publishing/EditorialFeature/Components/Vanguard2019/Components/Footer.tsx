import { Box, Flex, Sans, Separator, Serif } from "@artsy/palette"
import React from "react"

export const VanguardFooter = () => {
  return (
    <Box background="white" pb={80}>
      <Box
        maxWidth={["100vw", 1000]}
        px={["10vw", "10vw", "10vw", "10vw", 4]}
        mx="auto"
      >
        <Separator mb={4} />
        <Flex flexDirection={["column", "row"]}>
          <Box pr={[0, 2]} width={["100%", "50%"]} pb={4}>
            <Sans size="3" weight="medium">
              Corrections
            </Sans>
            <Serif size="3" element="p">
              {corrections}
            </Serif>
          </Box>
          <Box pl={[0, 2]} width={["100%", "50%"]} pb={4}>
            <Sans size="3" weight="medium">
              Clarifications
            </Sans>
            <Serif size="3" element="p">
              {clarifications}
            </Serif>
          </Box>
        </Flex>
        <Separator mb={2} />
        <Serif size="3">
          Back to <a href="/articles">Artsy Editorial</a>
        </Serif>
      </Box>
    </Box>
  )
}

const corrections = `A previous version of this article referred to Aike gallery as
Aike-Dellarco; McArthur Binion’s show at the Mississippi Museum of
Art, originally scheduled for this fall, is not opening until 2020;
and Melike Kara is Kurdish-German not Turkish-German. The text has
been updated to reflect these changes.`

const clarifications = `While only Beyoncé was named as the buyer of a Derek Fordjour work at
Frieze New York, Jay-Z was actually the buyer; in order to fully
recognize the breadth of galleries that represent the featured
artists, additional gallery names have been added to the text for
Beatriz González, Vivian Suter, and Diane Simpson; while the text on
Henry Taylor previously referred to Eva Presenhuber as his dealer, it
has been updated to reflect that Blum & Poe has been primarily
representing the artist for the past decade; and mention of
SculptureCenter, organizers of Tishan Hsu’s 2020 show at the Hammer,
has been added to clarify their involvement with the exhibition.`
