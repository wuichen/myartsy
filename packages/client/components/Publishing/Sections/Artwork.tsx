import React from "react"
import styled from "styled-components"
import { resize } from "../../../Utils/resizer"
import { GLOBAL_IMAGE_QUALITY } from "../Constants"
import { ArticleLayout, SectionLayout } from "../Typings"
import { ArtworkCaption } from "./ArtworkCaption"
import { ImageWrapper } from "./ImageWrapper"

export interface ArtworkProps {
  artwork: any
  color?: string
  editing?: boolean
  layout?: ArticleLayout
  sectionLayout?: SectionLayout
  linked?: boolean
  width?: string | number
  height?: string | number
  slideshowIndex?: number
}

export class Artwork extends React.PureComponent<ArtworkProps> {
  static defaultProps = {
    linked: true,
    width: "100%",
    height: "auto",
  }

  render() {
    const {
      artwork,
      children,
      editing,
      linked,
      height,
      width,
      layout,
      slideshowIndex,
    } = this.props
    const src = resize(artwork.image, {
      width: 1200,
      quality: GLOBAL_IMAGE_QUALITY,
    })

    const Image = () => (
      <ImageWrapper
        editing={editing}
        layout={layout}
        linked={linked}
        src={src}
        width={width}
        height={height}
        alt={artwork.title}
        slideshowIndex={slideshowIndex}
      />
    )

    return (
      <ArtworkContainer>
        {linked ? (
          <ArtworkImageLink href={`/artwork/${artwork.slug}`}>
            <Image />
          </ArtworkImageLink>
        ) : (
          <Image />
        )}

        <ArtworkCaption {...this.props} />
        {children}
      </ArtworkContainer>
    )
  }
}

const ArtworkImageLink = styled.a`
  text-decoration: none;
`

const ArtworkContainer = styled.div`
  display: flex;
  flex-direction: column;
`
