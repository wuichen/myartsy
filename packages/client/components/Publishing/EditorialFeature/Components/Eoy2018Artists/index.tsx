import { Box, color, Flex, media, Sans, Serif, space } from "@artsy/palette"
import { compact, find, map } from "lodash"
import React from "react"
import styled from "styled-components"

import { unica } from "Assets/Fonts"
import { Byline, BylineContainer } from "Components/Publishing/Byline/Byline"
import { EditorialFeaturesProps } from "Components/Publishing/EditorialFeature/EditorialFeature"
import { Nav, NavContainer } from "Components/Publishing/Nav/Nav"
import { ArticleCardContainer } from "Components/Publishing/RelatedArticles/ArticleCards/ArticleCard"
import {
  ArticleCards,
  ArticlesWrapper,
} from "Components/Publishing/RelatedArticles/ArticleCards/ArticleCards"
import {
  FullLabel,
  ImageSetContainer,
  ImageSetPreview,
  ImgContainer,
} from "Components/Publishing/Sections/ImageSetPreview"
import {
  LabelWrapper,
  SlideshowCta,
  SlideshowTitle,
} from "Components/Publishing/Sections/ImageSetPreview/ImageSetLabel"
import { StyledText } from "Components/Publishing/Sections/StyledText"
import { Text } from "Components/Publishing/Sections/Text"
import { resize } from "Utils/resizer"
import { Eoy2018ArticleHeader } from "./ArticleHeader"

export class Eoy2018Artists extends React.Component<EditorialFeaturesProps> {
  getHeaderSections = () => {
    const { sections } = this.props.article
    const headers = []

    sections.map((section, index) => {
      let headerSection
      const sectionAfter = sections[index + 1]
      const sectionAfterisImage =
        sectionAfter && sectionAfter.type === "image_collection"

      if (this.sectionIsHeader(section)) {
        headerSection = {
          index,
          section,
        }
        if (sectionAfterisImage) {
          headerSection.imageSection = sectionAfter
        }
        headers.push(headerSection)
      }
    })
    return headers
  }

  sectionIsHeader = section => {
    const isText = section.type === "text"
    const isHeader = isText && section.body.includes("<h1>")
    return isHeader
  }

  sectionArtistHeader = (section, i) => {
    const headerSections = this.getHeaderSections()
    const { imageSection } = find(headerSections, ["section", section])
    const src = imageSection && imageSection.images[0].url
    const caption = imageSection && imageSection.images[0].caption

    return (
      <ArtistHeaderSection
        key={i}
        mb={40}
        flexDirection={["column", "column", "row", "row"]}
      >
        <ArtistHeaderTitle dangerouslySetInnerHTML={{ __html: section.body }} />
        <ArtistHeaderImg
          src={src && resize(src, { width: 700 })}
          alignItems="flex-end"
        >
          {caption && (
            <Sans size="1" color="white" px={10} py={5}>
              <div dangerouslySetInnerHTML={{ __html: caption }} />
            </Sans>
          )}
        </ArtistHeaderImg>
      </ArtistHeaderSection>
    )
  }

  sectionText = (section, i) => {
    const {
      article: { sections, layout },
      showTooltips,
    } = this.props
    const isChapterStart =
      sections[i - 1] && sections[i - 1].type === "image_collection"

    return (
      <Box maxWidth={["100%", "75%", "75%"]} ml="auto" px={[20, 0]} key={i}>
        <TextSection size="5" pb={[40, 60]} isChapterStart={isChapterStart}>
          <Text
            html={section.body}
            layout={layout}
            showTooltips={showTooltips}
          />
        </TextSection>
      </Box>
    )
  }

  sectionImageSet = (section, i) => {
    return (
      <ImageSetWrapper
        key={i}
        mb={60}
        maxWidth={["100%", "100%", "100%", "75%"]}
      >
        <ImageSetPreview section={section}>
          <CaptionWrapper size={["3", "4"]}>
            <ImageSetCaption
              dangerouslySetInnerHTML={{ __html: section.images[0].caption }}
            />
          </CaptionWrapper>
        </ImageSetPreview>
      </ImageSetWrapper>
    )
  }

  getSections = () => {
    const {
      article: { sections },
    } = this.props
    const headerSections = this.getHeaderSections()
    const headerTextIndexes = map(headerSections, "index")
    const headerImages = map(headerSections, "image")

    const renderedSections = sections.map((section, i) => {
      const isHeader = headerTextIndexes.includes(i)
      const isHeaderImage = headerImages.includes(section)

      if (isHeader) {
        return this.sectionArtistHeader(section, i)
      } else if (isHeaderImage) {
        return null
      } else {
        if (section.type === "text" && i !== 0) {
          return this.sectionText(section, i)
        } else if (section.type === "image_set") {
          return this.sectionImageSet(section, i)
        } else {
          return null
        }
      }
    })
    return renderedSections
  }

  render() {
    const { article, isMobile, isTablet, isTest } = this.props
    const introText = this.sectionText(article.sections[0], 0)
    const headerImages = map(
      compact(map(this.getHeaderSections(), "imageSection")),
      "images"
    )

    return (
      <AricleWrapper>
        <Nav canFix color={color("black100")} backgroundColor="white">
          <NavBorder />
        </Nav>

        <Box px={[10, 10, 55]} maxWidth={1600} mx="auto">
          <Eoy2018ArticleHeader
            images={headerImages}
            isMobile={isMobile}
            isTablet={isTablet}
            isTest={isTest}
          />

          <ArticleContent py={40}>
            <IntroSection
              alignItems="flex-start"
              flexDirection={["column", "row"]}
              pl={[0, 20]}
            >
              <Byline article={article} />
              {introText}
            </IntroSection>
            {this.getSections()}
          </ArticleContent>
        </Box>

        {article.relatedArticles && (
          <Box px={[10, 10, 55]} mb={40} mx="auto">
            <RelatedArticleWrapper>
              <ArticleCards relatedArticles={article.relatedArticles} />
            </RelatedArticleWrapper>
          </Box>
        )}
      </AricleWrapper>
    )
  }
}

const BORDER_WIDTH = 6

const AricleWrapper = styled.div`
  ${NavContainer} {
    border-bottom: none;
  }
`

const NavBorder = styled.div`
  border-bottom: ${BORDER_WIDTH}px solid ${color("purple100")};
  position: absolute;
  top: 100%;
  left: 55px;
  right: 55px;
  max-width: calc(1600px - 110px);
  margin: 0 auto;

  ${media.md`
    left: 10px;
    right: 10px;
  `};
`

const ArticleContent = styled(Box)`
  border-left: ${BORDER_WIDTH}px solid ${color("purple100")};
  border-bottom: ${BORDER_WIDTH}px solid ${color("purple100")};

  blockquote {
    ${unica("s34")};
    line-height: 1.3em;
    margin: 0;
    font-weight: inherit;

    ${media.sm`
      ${unica("s25")};
      line-height: 1.3em;
    `};
  }
`

export const ArtistHeaderSection = styled(Flex)`
  height: 60vh;
  min-height: 450px;
  border: ${BORDER_WIDTH}px solid ${color("purple100")};
  border-left-width: 0;

  ${media.md`
    height: fit-content;
  `};
`

const ArtistHeaderTitle = styled.div`
  flex: 1;
  border-right: ${BORDER_WIDTH}px solid ${color("purple100")};
  overflow: hidden;
  min-height: min-content;

  h1 {
    ${unica("s65")};
    width: 100%;
    height: 50%;
    min-height: fit-content;
    padding: ${space(2)}px;
    border-bottom: ${BORDER_WIDTH}px solid ${color("purple100")};
    margin: 0;
    font-weight: inherit;

    ${media.md`
      width: 60%;
      float: right;
      height: 100%;
      border-bottom: none;
      border-left: ${BORDER_WIDTH}px solid ${color("purple100")};
    `};

    ${media.xs`
      ${unica("s45")};
      border-bottom: ${BORDER_WIDTH}px solid ${color("purple100")};
      border-left: none;
      width: 100%;
      height: 50%;
      float: none;
    `};
  }

  h2 {
    ${unica("s25")};
    width: 50%;
    height: 50%;
    min-height: fit-content;
    display: inline-flex;
    padding: ${space(2)}px;
    margin: 0;
    font-weight: inherit;

    &:last-child {
      border-left: ${BORDER_WIDTH}px solid ${color("purple100")};
    }

    ${media.md`
      width: 40%;
      float: left;
      &:last-child {
        border-left: none;
        border-top: ${BORDER_WIDTH}px solid ${color("purple100")};
      }
    `};

    ${media.xs`
      ${unica("s19")};
      width: 50%;
      float: none;
      &:last-child {
        border-left: ${BORDER_WIDTH}px solid ${color("purple100")};
        border-top: none;
      }
    `};
  }

  ${media.md`
    height: 65vh;
    border-right: none;
    border-bottom: ${BORDER_WIDTH}px solid ${color("purple100")};
  `};
`

const ArtistHeaderImg = styled(Flex)<{ src?: string }>`
  flex: 1;
  background: ${color("purple100")};

  p {
    text-shadow: 0 0 5px black;
    opacity: 0.6;
  }

  ${props =>
    props.src &&
    `
    background: url(${props.src});
    background-size: cover;
    background-position: center;
  `};

  ${media.md`
    min-height: 70vw;
  `};
`

const IntroSection = styled(Flex)`
  ${BylineContainer} {
    flex-direction: column;
    align-items: flex-start;
    ${media.md`
      padding: 0 20px 20px;
    `};
  }
`

const TextSection = styled(Serif)<{ isChapterStart?: boolean }>`
  ${StyledText} {
    blockquote {
      padding: 0;
    }

    p,
    .paragraph {
      font-size: 24px;
      text-indent: 2em;
      padding: 0;

      ${props =>
        props.isChapterStart &&
        `
        &:first-child {
          text-indent: 0;
        }
    `};
    }
  }
`

const CaptionWrapper = styled(Sans)`
  flex: 1;

  ${media.sm`
    min-width: 50%;
  `};
`

const ImageSetCaption = styled.div`
  height: 100%;
  padding: ${space(2)}px;
  background: white;
  color: black;
  border-bottom: ${BORDER_WIDTH}px solid ${color("purple100")};

  ${media.sm`
    border-bottom: none;
    border-right: ${BORDER_WIDTH}px solid ${color("purple100")};
  `};
`

const ImageSetWrapper = styled(Box)`
  border: ${BORDER_WIDTH}px solid ${color("purple100")};
  border-left: none;

  ${ImageSetContainer} {
    display: flex;
    flex-direction: row-reverse;

    ${media.sm`
      flex-direction: column-reverse;
    `};
  }

  ${FullLabel} {
    position: relative;
    display: flex;
    flex-direction: column-reverse;
    flex: 1;
    left: auto;
    bottom: auto;
    max-width: inherit;
    border-radius: 0;
    box-shadow: none;

    &:hover {
      background-color: ${color("purple100")};
    }
    ${media.sm`
      flex-direction: row-reverse;

      ${LabelWrapper} {
        max-width: 50%;
      }
    `};
  }

  ${SlideshowTitle} {
    display: none;
  }

  ${SlideshowCta} {
    flex-direction: column;

    > div {
      padding: 0;
    }
  }

  ${ImgContainer} {
    flex: 3;
    border-right: ${BORDER_WIDTH}px solid ${color("purple100")};

    img {
      object-fit: cover;
      object-position: center;
      height: 100%;
    }

    ${media.sm`
      border-right: none;
      border-bottom: ${BORDER_WIDTH}px solid ${color("purple100")};
    `};
  }
`

const RelatedArticleWrapper = styled.div`
  border-right: ${BORDER_WIDTH}px solid ${color("purple100")};
  border-bottom: ${BORDER_WIDTH}px solid ${color("purple100")};

  ${ArticlesWrapper} {
    margin-bottom: 0;
    margin-top: 0;
  }

  ${ArticleCardContainer} {
    border: none;
  }
`
