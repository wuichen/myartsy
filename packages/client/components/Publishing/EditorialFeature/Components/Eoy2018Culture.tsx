import { Box, color, Flex, Sans, Serif } from "@artsy/palette"
import { unica } from "Assets/Fonts"
import { media, pMedia } from "Components/Helpers"
import { Author, StyledAuthor } from "Components/Publishing/Byline/Author"
import { Date, DateContainer } from "Components/Publishing/Byline/Date"
import { Share } from "Components/Publishing/Byline/Share"
import { getArticleFullHref } from "Components/Publishing/Constants"
import { EditorialFeaturesProps } from "Components/Publishing/EditorialFeature/EditorialFeature"
import { Nav, NavContainer } from "Components/Publishing/Nav/Nav"
import { ArticleCards } from "Components/Publishing/RelatedArticles/ArticleCards/ArticleCards"
import { CaptionContainer } from "Components/Publishing/Sections/Caption"
import {
  ImageCollection,
  ImageCollectionItem,
} from "Components/Publishing/Sections/ImageCollection"
import {
  ImageSetContainer,
  ImageSetPreview,
} from "Components/Publishing/Sections/ImageSetPreview"
import {
  SlideshowCta,
  SlideshowTitle,
} from "Components/Publishing/Sections/ImageSetPreview/ImageSetLabel"
import { SocialEmbed } from "Components/Publishing/Sections/SocialEmbed"
import { StyledText } from "Components/Publishing/Sections/StyledText"
import { Text } from "Components/Publishing/Sections/Text"
import { ToolTipContainer } from "Components/Publishing/ToolTip/ToolTip"
import { SectionData } from "Components/Publishing/Typings"
import React from "react"
import Waypoint from "react-waypoint"
import styled from "styled-components"

interface State {
  stickyHeader: SectionData | null
}

export class Eoy2018Culture extends React.Component<
  EditorialFeaturesProps,
  State
> {
  state = {
    stickyHeader: null,
  }

  sectionText = (section, i, isDark) => {
    const {
      showTooltips,
      article: { layout },
    } = this.props
    return (
      <TextContainer key={i} px={[10, 10, 55]} py={30} mx="auto">
        <Box width={["100%", "100%", "70%"]} mx="auto" maxWidth={1000}>
          <Serif size="5">
            <Text
              html={section.body}
              layout={layout}
              showTooltips={showTooltips}
              color={isDark ? "white" : color("black100")}
            />
          </Serif>
        </Box>
      </TextContainer>
    )
  }

  sectionHeaderText = (section, i) => {
    return (
      <SectionHeader key={i} pt={10} pb={[10, 10, 20]} px={[10, 10, 55]}>
        <div dangerouslySetInnerHTML={{ __html: section.body }} />
      </SectionHeader>
    )
  }

  sectionSocialEmbed = (section, i) => {
    return (
      <SectionWrapper key={i} px={[10, 10, 55]} py={40}>
        <SocialEmbed section={section} />
      </SectionWrapper>
    )
  }

  sectionImageCollection = (section, i) => {
    return (
      <ImageWrapper key={i}>
        <ImageCollection
          sectionLayout={section.layout}
          articleLayout="feature"
          images={section.images}
          targetHeight={500}
          gutter={10}
        />
      </ImageWrapper>
    )
  }

  sectionImageSet = (section, i) => {
    return (
      <ImageSetWrapper key={i} px={[0, 0, 55]}>
        <ImageSetPreview section={section} />
      </ImageSetWrapper>
    )
  }

  sectionIsHeader = section => {
    const isText = section.type === "text"
    const isHeader = isText && section.body.includes("<h1>")
    return isHeader
  }

  makeSectionArrays = sections => {
    const chapters = []
    let currentChapter = null

    sections.map((section, i) => {
      if (this.sectionIsHeader(section)) {
        if (currentChapter === null) {
          currentChapter = 0
        } else {
          currentChapter += 1
        }
      }

      if (i !== 0) {
        if (chapters.length && chapters[currentChapter]) {
          chapters[currentChapter].push(section)
        } else {
          chapters[currentChapter] = [section]
        }
      }
    })
    return chapters
  }

  getSections = (sections, isDark) => {
    return sections.map((section, i) => {
      if (this.sectionIsHeader(section)) {
        return this.sectionHeaderText(section, i)
      } else {
        switch (section.type) {
          case "text": {
            return this.sectionText(section, i, isDark)
          }
          case "image_collection": {
            return this.sectionImageCollection(section, i)
          }
          case "image_set": {
            return this.sectionImageSet(section, i)
          }
          case "social_embed": {
            return this.sectionSocialEmbed(section, i)
          }
          default: {
            return null
          }
        }
      }
    })
  }

  render() {
    const { article } = this.props
    const chapters = this.makeSectionArrays(article.sections)
    const introText =
      article.sections && article.sections[0] && article.sections[0].body

    return (
      <ArticleWrapper>
        <Nav canFix color="black" backgroundColor="white">
          {this.state.stickyHeader && (
            <StickyHeader>
              {this.sectionHeaderText(this.state.stickyHeader, 0)}
            </StickyHeader>
          )}
        </Nav>
        <SectionWrapper>
          <ArticleTitle
            size="16"
            weight="medium"
            px={[10, 10, 55]}
            pb={10}
            pt={20}
          >
            <div>Year In Visual Culture</div>
            <div>2018</div>
          </ArticleTitle>
          <HeaderImg src={article.hero_section.url} />
        </SectionWrapper>

        <TextContainer px={[10, 10, 55]} py={30} mx="auto">
          <Box maxWidth={1400} mx="auto">
            <Flex
              pb={20}
              alignItems="center"
              flexDirection={["row-reverse", "row"]}
            >
              <Box width={["100%", "100%", "15%"]}>
                <Share
                  url={getArticleFullHref(article.slug)}
                  title={article.social_title || article.thumbnail_title}
                />
              </Box>
              <Flex width={["100%", "100%", "70%"]} maxWidth={1000}>
                <Author authors={article.authors} />
                <Date date={article.published_at} />
              </Flex>
            </Flex>

            <Box width={["100%", "100%", "70%"]} maxWidth={1000} mx="auto">
              {introText && (
                <div dangerouslySetInnerHTML={{ __html: introText }} />
              )}
            </Box>
          </Box>
          <Waypoint
            onEnter={({ previousPosition }) => {
              if (previousPosition === "above") {
                this.setState({ stickyHeader: null })
              }
            }}
          />
        </TextContainer>

        {chapters.map((chapter, i) => {
          const isDark = i % 2 === 0
          return (
            <ChapterWrapper isDark={isDark} key={i}>
              <Waypoint
                onLeave={({ currentPosition, previousPosition }) => {
                  if (
                    previousPosition === "inside" &&
                    currentPosition === "above"
                  ) {
                    this.setState({ stickyHeader: chapter[0] })
                  }
                }}
              />
              {this.getSections(chapter, isDark)}
              <Waypoint
                onEnter={({ previousPosition }) => {
                  if (previousPosition === "above") {
                    this.setState({ stickyHeader: chapter[0] })
                  }
                }}
              />
            </ChapterWrapper>
          )
        })}

        {article.relatedArticles && (
          <Box px={[10, 10, 55]} my={40} mx="auto">
            <ArticleCards relatedArticles={article.relatedArticles} />
          </Box>
        )}
      </ArticleWrapper>
    )
  }
}

const BORDER_WIDTH = 6

const ArticleWrapper = styled.div`
  ${NavContainer} {
    border-bottom-width: ${BORDER_WIDTH}px;
    border-top: ${BORDER_WIDTH}px solid black;
  }
  ${DateContainer} {
    margin-top: 0;
  }
  ${StyledAuthor} {
    margin-top: 0;

    ${media.xs`
      width: max-content;
    `};
  }
`

const ChapterWrapper = styled.div<{ isDark?: boolean }>`
  ${props =>
    props.isDark &&
    `
      color: white;
      background-color: ${color("black100")};
  `};

  ${ToolTipContainer} {
    color: ${color("black100")};
  }
`

const SectionWrapper = styled(Box)`
  border-bottom: ${BORDER_WIDTH}px solid;
  position: relative;
`

const TextContainer = styled(SectionWrapper)`
  blockquote {
    ${unica("s34")};
    padding: 20px 0;
    line-height: 1.25em;
    margin: 0;
    font-weight: inherit;
  }

  ${StyledText} {
    p,
    .paragraph {
      font-size: 24px;
      text-indent: 2em;
      padding: 0;

      &:first-child {
        text-indent: 0;
      }
    }
  }

  ${media.sm`
    blockquote {
      font-size: 24px;
    }
  `};
`

const ArticleTitle = styled(Sans)`
  text-transform: uppercase;
  border-bottom: ${BORDER_WIDTH}px solid;
  font-size: 7.5vw;
  line-height: 1em;
`

const SectionHeader = styled(Box)<{ isDark?: boolean }>`
  border-bottom: ${BORDER_WIDTH}px solid;

  > div {
    max-width: 1600px;
    margin: 0 auto;
    width: 100%;
    display: flex;
    align-items: flex-end;
  }

  h1 {
    ${unica("s65", "medium")};
    text-transform: uppercase;
    flex: 2;
    margin: 0;
    font-weight: inherit;
  }
  h2 {
    font-size: 50px;
    width: fit-content;
    line-height: 1em;
    text-align: right;
    margin: 0;
    font-weight: inherit;
  }

  ${media.md`
    h1 {
      font-size: 45px;;
    }
    h2 {
      font-size: 30px;
    }
  `};

  ${media.xs`
    h1 {
      font-size: 30px;;
    }
    h2 {
      font-size: 20px;
    }
`};
`

const HeaderImg = styled.img`
  height: calc(100vh - 50px);
  min-height: 400px;
  object-fit: cover;
  object-position: center;
  width: 100%;
`

const ImageWrapper = styled(SectionWrapper)`
  img {
    max-height: calc(100vh - 50px);
    min-height: 400px;
    object-fit: cover;
    object-position: center;

    ${media.md`
      min-height: unset;
    `};
  }

  ${ImageCollectionItem} {
    ${pMedia.xs`
      margin-bottom: 0;
      min-height: unset;
    `};
  }

  ${CaptionContainer} {
    position: absolute;
    bottom: 0;
    left: 5px;

    p {
      font-size: 10px;
      color: white;
      opacity: 0.6;
      text-shadow: 0 0 5px black;
    }
  }
`
const ImageSetWrapper = styled(SectionWrapper)`
  ${ImageSetContainer} {
    width: 70%;
    max-width: 1000px;
    margin: 0 auto;
    color: black;

    img {
      width: 100%;
    }

    ${SlideshowTitle} {
      display: none;
    }
    ${SlideshowCta} {
      flex-direction: column;
      > div {
        padding: 0;
      }
    }
    ${media.md`
      width: 100%;
    `};
  }
`

const StickyHeader = styled.div`
  position: absolute;
  top: 100%;
  width: 100%;
  background: white;
  border-top: ${BORDER_WIDTH}px solid;

  h1 {
    font-size: 40px;
  }
  h2 {
    font-size: 35px;
  }

  ${SectionHeader} {
    padding-bottom: 10px;
    padding-left: 10px;
    padding-right: 10px;
    padding-top: 14px;
  }

  ${media.sm`
    h1 {
      font-size: 30px;;
    }
    h2 {
      font-size: 20px;
    }
  `};
`
