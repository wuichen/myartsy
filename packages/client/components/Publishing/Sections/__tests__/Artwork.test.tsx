// tslint:disable:no-shadowed-variable

import { Images } from "Components/Publishing/Fixtures/Components"
import {
  EditableChild,
  WrapperWithFullscreenContext,
} from "Components/Publishing/Fixtures/Helpers"
import { mount } from "enzyme"
import "jest-styled-components"
import React from "react"
import renderer from "react-test-renderer"
import { Artwork } from "../Artwork"
import { ViewFullscreen } from "../ViewFullscreen"

jest.mock("react-lines-ellipsis/lib/html", () => {
  const React = require("react")
  return () => <div />
})

jest.mock("react-dom/server", () => ({
  renderToStaticMarkup: x => x,
}))

describe("Artwork", () => {
  const renderSnapshot = props => {
    return renderer
      .create(WrapperWithFullscreenContext(<Artwork {...props} />))
      .toJSON()
  }

  const mountWrapper = props => {
    return mount(WrapperWithFullscreenContext(<Artwork {...props} />))
  }

  let props
  beforeEach(() => {
    props = {}
  })

  describe("snapshots", () => {
    it("renders properly", () => {
      props = {
        artwork: Images[0],
      }
      const artwork = renderSnapshot(props)
      expect(artwork).toMatchSnapshot()
    })
  })

  describe("unit", () => {
    it("renders a fullscreen button if linked", () => {
      props = {
        artwork: Images[0],
        linked: true,
      }
      const component = mountWrapper(props)
      expect(component.find(ViewFullscreen).length).toBe(1)
    })

    it("does not render a fullscreen button if not linked", () => {
      props = {
        artwork: Images[0],
        linked: false,
      }
      const component = mountWrapper(props)
      expect(component.find(ViewFullscreen).length).toBe(0)
    })

    it("renders a child if present", () => {
      const component = mount(
        WrapperWithFullscreenContext(
          <Artwork artwork={Images[0]}>
            {EditableChild("A React child.")}
          </Artwork>
        )
      )
      expect(component.text()).toMatch("A React child.")
    })

    it("adds imagesLoaded class if props.editing", () => {
      props = {
        artwork: Images[0],
        editing: true,
      }
      const component = mountWrapper(props)
      expect(component.html()).toMatch("BlockImage__container image-loaded")
    })

    it("does not add imagesLoaded class if props.editing is false", () => {
      props = {
        artwork: Images[0],
      }
      const component = mountWrapper(props)
      expect(component.html()).not.toMatch("BlockImage__container image-loaded")
    })
  })
})
