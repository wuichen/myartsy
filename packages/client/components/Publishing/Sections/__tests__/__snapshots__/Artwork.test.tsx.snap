// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artwork snapshots renders properly 1`] = `
.c8 {
  font-family: Unica77LLWebRegular,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #C2C2C2;
}

.c9 {
  margin-right: 30px;
}

.c10 {
  white-space: nowrap;
}

.c7 {
  padding: 0;
  margin-top: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c7 a {
  color: #C2C2C2;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c7 .title,
.c7 .title a {
  font-style: italic;
}

.c6 {
  position: absolute;
  bottom: 0;
  right: 0;
  margin: 10px;
  width: 25px;
  height: 25px;
  cursor: pointer;
  opacity: 0.6;
  -webkit-transition: opacity 0.3s;
  transition: opacity 0.3s;
}

.c6:hover {
  opacity: 1;
}

.c5 {
  opacity: 0;
  -webkit-transition: opacity 0.3s;
  transition: opacity 0.3s;
}

.c2 {
  position: relative;
}

.c2:hover .c4 {
  opacity: 1;
}

.c2 .BlockImage__container {
  opacity: 0;
  -webkit-transition: opacity 1s;
  transition: opacity 1s;
}

.c2 .image-loaded {
  opacity: 1;
}

.c3 {
  display: block;
}

.c1 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

@media (max-width:600px) {
  .c7 {
    padding: 0 10px;
  }
}

@media (max-width:720px) {
  .c2 .c4 {
    opacity: 1;
  }
}

<div
  className="c0"
>
  <a
    className="c1"
    href="/artwork/fernando-botero-nude-on-the-beach"
  >
    <div
      className="c2"
    >
      <img
        alt="Nude on the Beach"
        className="BlockImage__container c3"
        height="auto"
        src="https://d7hftxdivxxvm.cloudfront.net?resize_to=width&src=https%3A%2F%2Fd32dm0rphc51dk.cloudfront.net%2F0aRUvnVgQKbQk5dj8xcCAg%2Flarger.jpg&width=1200&quality=80"
        width="100%"
      />
      <div
        className="c4 c5"
      >
        <div
          className="c6"
          onClick={[Function]}
        >
          <svg
            className="expand"
            version="1.1"
            viewBox="0 0 27 27"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fillRule="evenodd"
              stroke="none"
              strokeWidth="1"
            >
              <g>
                <circle
                  cx="13.9575195"
                  cy="13.0588379"
                  fill="#000000"
                  fillRule="nonzero"
                  r="13"
                />
                <g
                  stroke="#FFFFFF"
                  transform="translate(8.000000, 7.000000)"
                >
                  <g>
                    <g
                      transform="translate(0.382532, 0.382532)"
                    >
                      <polyline
                        points="4.78165571 0 11.1571966 0 11.1571966 6.37554094"
                      />
                      <polyline
                        points="-1.01881144e-06 4.78165537 6.37553992 4.78165537 6.37553992 11.1571963"
                        transform="translate(3.187769, 7.969426) rotate(-180.000000) translate(-3.187769, -7.969426) "
                      />
                      <path
                        d="M10.360254,0.796942618 L0.796942618,10.360254"
                        strokeLinecap="square"
                      />
                    </g>
                  </g>
                </g>
              </g>
            </g>
          </svg>
        </div>
      </div>
    </div>
  </a>
  <div>
    <div
      className="c7 c8"
      color="#C2C2C2"
      fontFamily="Unica77LLWebRegular, 'Helvetica Neue', Helvetica, Arial, sans-serif"
      fontSize="14px"
    >
      <span
        className="c9"
      >
        <span
          className="c10"
        >
          <a
            href="/artist/fernando-botero"
          >
            Fernando Botero
          </a>
        </span>
      </span>
      <div>
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": <span>
                <span
                  className="title"
                >
                  <a
                    href="/artwork/fernando-botero-nude-on-the-beach"
                  >
                    Nude on the Beach
                  </a>
                </span>
                <span>
                  , 
                </span>
                <span
                  className="date"
                >
                  2000
                </span>
              </span>,
            }
          }
        />
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": <span>
                <a
                  href="/gary-nader"
                >
                  Gary Nader
                </a>
                <span>
                  . 
                </span>
                <span
                  className="credit"
                >
                  Courtesy of Gary Nader
                </span>
              </span>,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
